services/payments/service.go

```diff
@@ -0,0 +1,180 @@
+package payments
+
+import (
+	"context"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/awslabs/amazon-qldb-driver-go/v3/qldbdriver"
+	"github.com/brave-intl/bat-go/libs/custodian/provider"
+	"github.com/brave-intl/bat-go/libs/nitro"
+	"github.com/hashicorp/vault/shamir"
+
+	"encoding/json"
+
+	awsutils "github.com/brave-intl/bat-go/libs/aws"
+	appctx "github.com/brave-intl/bat-go/libs/context"
+	"github.com/brave-intl/bat-go/libs/cryptography"
+	"github.com/brave-intl/bat-go/libs/logging"
+	appsrv "github.com/brave-intl/bat-go/libs/service"
+)
+
+// Service - struct definition of payments service
+type Service struct {
+	// concurrent safe
+	datastore  *qldbdriver.QLDBDriver
+	custodians map[string]provider.Custodian
+
+	baseCtx          context.Context
+	secretMgr        appsrv.SecretManager
+	keyShares        [][]byte
+	kmsDecryptKeyArn string
+}
+
+// createKMSKey creates the enclave kms key which is only decrypt capable with enclave attestation.
+func (s *Service) configureKMSKey(ctx context.Context) error {
+	// perform enclave attestation
+	nonce := make([]byte, 64)
+	_, err := rand.Read(nonce)
+	if err != nil {
+		return fmt.Errorf("failed to create nonce for attestation: %w", err)
+	}
+	document, err := nitro.Attest(nonce, nil, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create attestation document: %w", err)
+	}
+	var logger = logging.Logger(ctx, "payments.configureKMSKey")
+	logger.Debug().Msgf("document: %+v", document)
+
+	// get the aws configuration loaded
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to load aws configuration: %w", err)
+	}
+
+	// TODO: get the pcr values for the condition from the document ^^
+	imageSha384 := ""
+	pcr0 := ""
+	pcr1 := ""
+	pcr2 := ""
+
+	// get the secretsmanager id from ctx for the template
+	templateSecretID, ok := ctx.Value(appctx.EnclaveDecryptKeyTemplateSecretIDCTXKey).(string)
+	if !ok {
+		return errors.New("template secret id for enclave decrypt key not found on context")
+	}
+
+	smClient := secretsmanager.NewFromConfig(cfg)
+	o, err := smClient.GetSecretValue(ctx, &secretsmanager.GetSecretValueInput{
+		SecretId: aws.String(templateSecretID),
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to get key policy template from secrets manager: %w", err)
+	}
+
+	if o.SecretString == nil {
```

Comment:
Similarly can we define further what the relevance of this secretString is for with a slightly better name? It's not clear to me from reading the code what this SecretString is and why it needs to be loaded. Is this actually doing Bootstrap step 2?